src/lib.rs

```diff
@@ -15,6 +15,19 @@
 //! The `proto` module contains the standard UEFI protocols, which are normally provided
 //! by the various UEFI drivers and firmware layers.
 //!
+//! ## Optional Crate Features:
+//! - `alloc`: Enables functionality requiring the `alloc` crate from the Rust standard library.
```

Comment:
Minor nitpicks:
- Could you leave a blank line after the `##` and before the first list item, like above?
- Could you write the section title as `## Optional crate features`, to match the titles above?